prover/src/keccak_sponge/keccak_sponge_stark.rs

```diff
@@ -250,7 +250,7 @@ impl<F: RichField + Extendable<D>, const D: usize> KeccakSpongeStark<F, D> {
         rows
     }
 
-    fn generate_rows_for_op(&self, op: KeccakSpongeOp) -> Vec<[F; NUM_KECCAK_SPONGE_COLUMNS]> {
+    fn generate_rows_for_op(&self, op: &KeccakSpongeOp) -> Vec<[F; NUM_KECCAK_SPONGE_COLUMNS]> {
```

Comment:
it's better to use the op as a value directly, cause the op will be deref after this function is called, hence we don't need to drop the op manually.